### PR TITLE
Cleanup and restructure Keepers content.

### PR DESCRIPTION
**Pull request by dwightjl**
- Edit existing keepers content.
- Move Keepers code samples to the Keepers directory.
- Remove unnecessary newlines from the ends of Keepers code samples.
- Add some code samples where relevant.

### CODE REVIEW

File: docs/chainlink-keepers/best-practices.md [hunk 1]
```diff
@@ -8,81 +8,86 @@ whatsnext:
...
+```solidity Rinkeby
+{% include samples/Keepers/checkData.sol %}
+```
 
 ### `performData`
 
-The response from `checkUpkeep` gets passed as `performData` to `performUpkeep`. This allows you to perform complex and costly simulations with no gas cost, then identifying the subset of actions you’re ready to take based on the conditions met.
-
-With these two inputs, you have a highly flexible off-chain computation infrastructure that can perform precise actions on-chain, both of which are entirely programmable.
+The response from `checkUpkeep` passes as `performData` to the `performUpkeep` function. This allows you to perform complex and costly simulations with no gas cost. Then you can identify the subset of actions that you are ready to take based on the conditions that are met.
```
Comment (alexroan) [inline docs/chainlink-keepers/best-practices.md:29]:
```suggestion
The response from `checkUpkeep` is passed to the `performUpkeep` function as `performData`. This allows you to perform complex and costly simulations with no gas cost. Then you can identify the subset of actions that you are ready to take based on the conditions that are met.
```

File: docs/chainlink-keepers/best-practices.md [hunk 2]
```diff
@@ -8,81 +8,86 @@ whatsnext:
     'Network Overview': '/docs/chainlink-keepers/overview/',
   }
 ---
-Chainlink Keepers are a powerful tool for smart contract automation. These are some useful patterns and best practices that you can employ within your Keeper-compatible smart contract.
+Chainlink Keepers are a powerful tool for smart contract automation. This guide demonstrates some useful patterns and best practices that you can employ within Keeper-compatible smart contracts.
 
-It is important to understand the gas limits for on-chain execution and off-chain computation when developing your Keeper-compatible smart contract - you can review that in [How Chainlink Keepers Work](../overview/).
+It is important to understand the gas limits for on-chain execution and off-chain computation when developing your Keeper-compatible smart contract. To learn more about the economics of Chainlink Keepers, read [How Chainlink Keepers Work](../overview/).
```
Comment (alexroan) [inline docs/chainlink-keepers/best-practices.md:13]: The "How Chainlink Keepers Work" page is actually the "Network Overview" page. That strikes me as potentially confusing.

If there another way we could phrase this, or "Network Overview" to fit a little better?

File: docs/chainlink-keepers/best-practices.md [hunk 3]
```diff
@@ -8,81 +8,86 @@ whatsnext:
...
 
 ## Testing
 
 As with all smart contract testing, it is important to test the boundaries of your smart contract in order to ensure it operates as intended. Similarly, it is important to make sure your Keeper-compatible contract operates within the parameters of the `KeeperRegistry`.
 
-Your mission critical contracts should have all code paths tested and you should stress-test your contract to confirm the performance and correct operation of your use case under load or adversarial conditions. The Chainlink Keeper Network will continue to operate under stress, but so should your contract. [Reach out](https://forms.gle/WadxnzzjHPtta5Zd9) to us if you need any help, have questions, or feedback for improvement.
+Test all of your mission-critical contracts, and stress-test the contract to confirm the performance and correct operation of your use case under load and adversarial conditions. The Chainlink Keeper Network will continue to operate under stress, but so should your contract. [Reach out](https://forms.gle/WadxnzzjHPtta5Zd9) to us if you need help, have questions, or feedback for improvement.
 
 ## Gas Limits
```
Comment (alexroan) [inline docs/chainlink-keepers/best-practices.md:91]: Let's move the "Gas Limits" section above "Testing".

File: docs/chainlink-keepers/best-practices.md [hunk 4]
```diff
@@ -8,81 +8,86 @@ whatsnext:
     'Network Overview': '/docs/chainlink-keepers/overview/',
   }
 ---
-Chainlink Keepers are a powerful tool for smart contract automation. These are some useful patterns and best practices that you can employ within your Keeper-compatible smart contract.
+Chainlink Keepers are a powerful tool for smart contract automation. This guide demonstrates some useful patterns and best practices that you can employ within Keeper-compatible smart contracts.
 
-It is important to understand the gas limits for on-chain execution and off-chain computation when developing your Keeper-compatible smart contract - you can review that in [How Chainlink Keepers Work](../overview/).
+It is important to understand the gas limits for on-chain execution and off-chain computation when developing your Keeper-compatible smart contract. To learn more about the economics of Chainlink Keepers, read [How Chainlink Keepers Work](../overview/).
 
 ## Use `checkData` and `performData` arguments
```
Comment (alexroan) [inline docs/chainlink-keepers/best-practices.md:15]: Current H2 Headings:

- Use `checkData` and `performData` arguments
- Modifiers
- Patterns to Consider
- Testing
- Gas Limits

I think a better set would be:

- Functions
- Modifiers
- Useful Patterns
- Gas Limits
- Testing

File: docs/chainlink-keepers/introduction.md [hunk 5]
```diff
@@ -29,18 +29,19 @@ The Chainlink Keeper Network is a decentralized solution where independent Keepe
...
 > Our goal is to ensure flawless execution of your Upkeeps. To get the most out of Chainlink Keepers, we recommend you review these docs in full - understand their capabilities, usage patterns, best practices, and how to maintain the health of your upkeep.
 
 ## Onboarding Steps
 
-We strongly recommend to start by testing your Keeper-compatible contracts on testnet before moving to mainnet. At any point during your testing process, please [reach out to us](https://forms.gle/WadxnzzjHPtta5Zd9) if you have questions that have not been addressed, or take a look at our [Keepers Frequently Asked Questions](../faqs/).
+Start by testing your Keeper-compatible contracts on testnet before moving to mainnet. If you have questions, take a look at our [Keepers Frequently Asked Questions](../faqs/) page or [reach out to us](https://forms.gle/WadxnzzjHPtta5Zd9).
 
 1. Create a [Keeper compatible contract](../compatible-contracts/)
 1. Deploy your contract onto a supported testnet
 1. Get [LINK](../../link-token-contracts/#kovan) on the testnet
```
Comment (alexroan) [inline docs/chainlink-keepers/introduction.md:43]: It's cool how all these `1.`s get transformed into a list 😄

**Review by alexroan — CHANGES_REQUESTED** [review]
Nice work, pretty close! Just a few small changes requested.

File: docs/chainlink-keepers/best-practices.md (hunk 1 above)
Comment (dwightjl) [reply docs/chainlink-keepers/best-practices.md]: Updated in latest push.

File: docs/chainlink-keepers/best-practices.md (hunk 4 above)
Comment (dwightjl) [reply docs/chainlink-keepers/best-practices.md]: Updated in latest push.

File: docs/chainlink-keepers/best-practices.md (hunk 3 above)
Comment (dwightjl) [reply docs/chainlink-keepers/best-practices.md]: Updated in latest push.

File: docs/chainlink-keepers/best-practices.md (hunk 2 above)
Comment (dwightjl) [reply docs/chainlink-keepers/best-practices.md]: Updated in latest push.

**Review by alexroan — APPROVED** [review]

**Pull request merged**